node/src/main/scala/org/bitcoins/node/SpvNode.scala

```diff
@@ -49,6 +48,36 @@ case class SpvNode(
     PeerMessageSender(client, nodeAppConfig.network)
   }
 
+  /** Updates our bloom filter to match the given TX
+    *
+    * @return SPV node with the updated bloom filter
+    */
+  def updateBloomFilter(transaction: Transaction): SpvNode = {
+    logger.info(s"Updating bloom filter with transaction=${transaction.txIdBE}")
+    val newBloom = bloomFilter.update(transaction)
+
+    // we could send filteradd messages, but we would
+    // then need to calculate all the new elements in
+    // the filter. this is easier:-)
+    peerMsgSender.sendFilterClearMessage()
+    peerMsgSender.sendFilterLoadMessage(newBloom)
```

Comment:
Is this comment true? Wouldn't it be better to have the `bloomFilter` have a method that let you add things without reconstructing from scratch and then also add those same things (via a single aggregate Bloom filter) with a message rather than sending two messages?

Reply:
The problem here is that `BloomFilter#update(transaction)` returns the new bloom filter, and not the elements we need to insert. If it returned the latter we could send just those elements. 

I don't think this matters too much, although I see your point

Reply:
We probably should add the necessary stuff to the `BloomFilter` api to accomplish this. #667